src/config.rs

```diff
@@ -94,36 +94,26 @@ impl LocalizedConfig {
         config.map(LocalizedConfig::new).transpose()
     }
 
-    /// Read an environment variable from the `[env]` section in this `.cargo/config.toml`.
+    /// Propagate environment variables from this `.cargo/config.toml` to the process environment
+    /// using [`std::env::set_var()`].
     ///
-    /// It is interpreted as path and canonicalized relative to [`Self::workspace`] if
-    /// [`EnvOption::Value::relative`] is set.
-    ///
-    /// Process environment variables (from [`std::env::var()`]) have [precedence]
-    /// unless [`EnvOption::Value::force`] is set. This value is also returned if
-    /// the given key was not set under `[env]`.
-    ///
-    /// [precedence]: https://doc.rust-lang.org/cargo/reference/config.html#env
-    pub fn resolve_env(&self, key: &str) -> Result<Cow<'_, str>> {
-        let config_var = self.config.env.as_ref().and_then(|env| env.get(key));
-
-        // Environment variables always have precedence unless
-        // the extended format is used to set `force = true`:
-        if let Some(env_option @ EnvOption::Value { force: true, .. }) = config_var {
-            // Errors iresolving (canonicalizing, really) the config variable take precedence, too:
-            return env_option.resolve_value(&self.workspace);
-        }
-
-        let process_var = std::env::var(key);
-        if process_var != Err(VarError::NotPresent) {
-            // Errors from env::var() also have precedence here:
-            return Ok(process_var?.into());
```

Comment:
This was a public function, not sure if we should keep it (and the tests associated with it)?

If we remove it (saves some maintenance overhead, I don't think the precedence rules still make sense when calling this function after `set_env_vars()` has been used anyway), this is a breaking change.